### ISSUE TITLE
Derby date time arithmetic uses unsupported sql_tsi_milli_second

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#10254]
When performing date time arithmetic with Derby, HSQLDB's `SQL_TSI_MILLI_SECOND` is being generated, which is not supported. `SQL_TSI_FRAC_SECOND` should be generated instead.

It can be reproduced on https://www.jooq.org/translate, when translating this to Derby:

```
-- Input
current_timestamp - interval '1' second

-- Output
{fn timestampadd(sql_tsi_second, -1, {fn timestampadd(sql_tsi_milli_second, 0, current_timestamp) }) }
```

**Comment by lukaseder** [comment]
This is a regression introduced in jOOQ 3.12: https://github.com/jOOQ/jOOQ/issues/8065

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.14.0. Backport scheduled for 3.13.3 (#10255)
